File: python/problem1441.py
class Solution:
    def buildArray(self, target: List[int], n: int) -> List[str]:
        result = []
        count = 1
        for i in target:
            if i == count:
                result.append("Push")
            else:
                while count < i:
                    count += 1
                    result.append("Push")
                    result.append("Pop")
                result.append("Push")
            count += 1
        return result

File: Java/problem1441.java
class Solution {
    public List<String> buildArray(int[] target, int n) {
        List<String> result = new LinkedList<>();
        int count = 1;
        for (int i = 0; i < target.length; ++i) {
            if (target[i] == count) {
                ++count;
                result.add("Push");
            }
            else {
                while (count < target[i]) {
                    result.add("Push");
                    result.add("Pop");
                    ++count;
                }
                result.add("Push");
                ++count;
            }
        }
        return result;
    }
}
